app/scripts/controllers/transactions/index.js

```diff
@@ -331,6 +331,137 @@ export default class TransactionController extends EventEmitter {
     });
   }
 
+  // ====================================================================================================================================================
+
+  /**
+   * @param {number} txId
+   * @returns {TransactionMeta} the txMeta who matches the given id if none found
+   * for the network returns undefined
+   */
+  _getTransaction(txId) {
+    const { transactions } = this.store.getState();
+    return transactions[txId];
+  }
+
+  _validateTransaction(txId) {
```

Comment:
I think it would be good if the name of this method reflected the specific purpose: to check if the transaction status is `unapproved`

Perhaps rename it to `_checkIfTxStatusIsUnapproved`